SYCL/ESIMD/api/functional/ctors/operator_logical_not.cpp

```diff
@@ -0,0 +1,154 @@
+//
+// Test for simd logical not operator.
+// The test creates source simd instance with reference data and invokes logical
+// not operator.
+// The test verifies that data from simd is not corrupted after calling logical
+// not operator, that logical not operator return type is as expected and
+// logical not operator result values is correct.
+
+#include "common.hpp"
+
+using namespace sycl::ext::intel::experimental::esimd;
+using namespace esimd_test::api::functional;
+
+// Descriptor class for the case of calling logical not operator.
+struct logical_not_operator {
+  static std::string get_description() { return "logical not"; }
+
+  template <typename DataT, int NumElems, typename FlagT>
+  static bool call_simd_ctor(const DataT *const ref_data, DataT *const out,
+                             FlagT *const logical_not_res_is_correct) {
+    auto simd_obj = simd<DataT, NumElems>();
+    simd_obj.copy_from(ref_data);
+    const auto logical_not_result = !simd_obj;
+
+    for (size_t i = 0; i < NumElems; ++i) {
+      *logical_not_res_is_correct &=
+          logical_not_result[i] == (ref_data[i] == 0);
+    }
+    simd_obj.copy_to(out);
+    return std::is_same_v<decltype(!simd_obj), simd_mask<NumElems>>;
+  }
+};
+
+// The main test routine.
+// Using functor class to be able to iterate over the pre-defined data types.
+template <typename DataT, typename DimT, typename TestCaseT> class run_test {
+  static constexpr int NumElems = DimT::value;
+
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type) {
+    bool passed = true;
+    const std::vector<DataT> ref_data = generate_ref_data<DataT, NumElems>();
+
+    // If current number of elements is equal to one, then run test with each
+    // one value from reference data.
+    // If current number of elements is greater than one, then run tests with
+    // whole reference data.
+    if constexpr (NumElems == 1) {
+      for (size_t i = 0; i < ref_data.size(); ++i) {
+        passed = run_verification(queue, {ref_data[i]}, data_type);
+      }
+    } else {
+      passed = run_verification(queue, ref_data, data_type);
+    }
+    return passed;
+  }
+
+private:
+  bool run_verification(sycl::queue &queue, const std::vector<DataT> &ref_data,
+                        const std::string &data_type) {
+    assert(ref_data.size() == NumElems &&
+           "Reference data size is not equal to the simd vector length.");
+
+    bool passed = true;
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> result(NumElems, allocator);
+    shared_vector<DataT> shared_ref_data(ref_data.begin(), ref_data.end(),
+                                         allocator);
+    // TODO replace using std::vector with using shared_element
+    shared_vector<int> logical_not_res_is_correct(1,
+                                                  shared_allocator<int>(queue));
+    shared_vector<int> logical_not_res_type_is_correct(
+        1, shared_allocator<int>(queue));
```

Comment:
```suggestion
    shared_vector<int> is_correct_result(1,
                                                  shared_allocator<int>(queue));
    shared_vector<int> is_correct_type(
        1, shared_allocator<int>(queue));
```
1) We are in `operator_logical_not.cpp` anyway. So in scope of this test there is no other operator.
2) You have `TestCaseT::get_description()` later, but use explicit `logical not operator` sometimes

Reply:
these variables was renamed [here](https://github.com/intel/llvm-test-suite/pull/784/commits/e868d819250aebcd33b836e14dd71cd62fb4e257#diff-eac54e76a314e24c2d279609017dedef6dab795a4af3a589567c94d08070bf55R87-R88)

Reply:
You are right, the `operator_result` fits much better